### ISSUE TITLE
hark: crash when clearing notifcations

### ISSUE THREAD

**Issue by timlucmiptev** [urbit/urbit#4201]
**Describe the bug**
In one particular channel, when I try to mark messages as unread, I get the following crash:
```
Uncaught (in promise) /app/hark-store/hoon:<[392 7].[392 39]>
/app/hark-store/hoon:<[390 7].[392 39]>
/app/hark-store/hoon:<[584 5].[584 39]>
/app/hark-store/hoon:<[583 3].[585 39]>
/app/hark-store/hoon:<[582 3].[585 39]>
/app/hark-store/hoon:<[388 7].[392 39]>
/app/hark-store/hoon:<[386 5].[393 7]>
/app/hark-store/hoon:<[385 5].[393 7]>
/app/hark-store/hoon:<[398 10].[398 45]>
/app/hark-store/hoon:<[398 5].[398 45]>
/app/hark-store/hoon:<[397 5].[398 45]>
/app/hark-store/hoon:<[481 7].[481 30]>
/app/hark-store/hoon:<[480 5].[482 35]>
/app/hark-store/hoon:<[478 5].[482 35]>
/app/hark-store/hoon:<[476 5].[482 35]>
/app/hark-store/hoon:<[475 9].[482 35]>
/app/hark-store/hoon:<[479 7].[479 21]>
/app/hark-store/hoon:<[478 5].[482 35]>
/app/hark-store/hoon:<[476 5].[482 35]>
/app/hark-store/hoon:<[475 9].[482 35]>
/app/hark-store/hoon:<[482 5].[482 35]>
/app/hark-store/hoon:<[480 5].[482 35]>
/app/hark-store/hoon:<[478 5].[482 35]>
/app/hark-store/hoon:<[476 5].[482 35]>
/app/hark-store/hoon:<[475 9].[482 35]>
/app/hark-store/hoon:<[479 7].[479 21]>
/app/hark-store/hoon:<[478 5].[482 35]>
/app/hark-store/hoon:<[476 5].[482 35]>
/app/hark-store/hoon:<[475 9].[482 35]>
/app/hark-store/hoon:<[475 5].[482 35]>
/app/hark-store/hoon:<[473 5].[482 35]>
/app/hark-store/hoon:<[468 7].[468 45]>
/app/hark-store/hoon:<[467 5].[469 37]>
/app/hark-store/hoon:<[466 5].[469 37]>
/app/hark-store/hoon:<[465 5].[469 37]>
/app/hark-store/hoon:<[461 11].[461 41]>
/app/hark-store/hoon:<[461 6].[461 41]>
/app/hark-store/hoon:<[461 5].[461 66]>
/app/hark-store/hoon:<[459 5].[461 66]>
/app/hark-store/hoon:<[458 5].[461 66]>
/app/hark-store/hoon:<[320 22].[320 39]>
/app/hark-store/hoon:<[314 5].[332 7]>
/app/hark-store/hoon:<[313 5].[332 7]>
/app/hark-store/hoon:<[274 10].[274 48]>
/app/hark-store/hoon:<[274 5].[274 48]>
/app/hark-store/hoon:<[273 5].[274 48]>
/app/hark-store/hoon:<[266 24].[266 59]>
/app/hark-store/hoon:<[265 5].[268 7]>
/app/hark-store/hoon:<[264 3].[269 15]>
/app/hark-store/hoon:<[263 3].[269 15]>
/app/hark-store/hoon:<[262 3].[275 5]>
/app/hark-store/hoon:<[261 3].[275 5]>
/sys/vane/gall/hoon:<[1.375 9].[1.375 37]>
```

**To Reproduce**
Have not been able to repro in other channels. This happens regardless of hard refreshes, etc.

**Desktop (please complete the following information):**
 - OS: MacOS 11.1
 - Browser: Brave
 - Base hash: 74m9v

**Comment by timlucmiptev** [comment]
This is the line in question:
https://github.com/urbit/urbit/blob/master/pkg/arvo/app/hark-store.hoon#L392
`?>(!=(read read.n) n(read read))`

I'm not super-familiar with `hark`, but it seems that the store got itself into an inconsistent state wrt marking messages as read and now can't get out.

**Comment by matildepark** [comment]
Fixed in #4262.
